flakey-test-tracking/README.md

```diff
@@ -0,0 +1,69 @@
+Optionally, FlakeTracker can skip known flakey tests by fetching the list of flakey test issues from GitHub before each test suite. FlakeTracker authenticates with the GitHub API as a GitHub App, which allows it to provide rich feedback on PRs via GitHub Checks.
+
+## Configuration
+
+You can configure FlakeTracker via the following environment variables.
+
+- `FLAKEY`
+  - When running in CI, set `FLAKEY=false` to disable FlakeTracker, which will be enabled by default.
+  - When running locally, set `FLAKEY=true` to enable FlakeTracker, which will be disabled by default.
+- `LOG_ALL_RETRY_ERRORS`
+  - Only relevant when flake tracker is enabled.
+  - Enables error logging after retries even for tests that never pass. Defaults to `false`.
+- `NUM_RETRIES`
+  - Only relevant when flake tracker is enabled.
+  - Specifies how many retries should be performed before a test is failed.
+  - If you want a "fail fast without flakes" option (let's say you know a bunch of tests will fail, and you want to skip all the flakey tests without retrying each failed test a bunch of times) you can accomplish this by setting `NUM_RETRIES=0`.
+- `SKIP_KNOWN_FLAKES`
+  - Only relevant when flake tracker is enabled and tests are run in CI.
+  - Set `SKIP_KNOWN_FLAKES=false` to disable the skipping of known flakes in CI. Defaults to `true`.
+- `FLAKES_FAIL_CHECK_SUITE`
+  - If true, new flakey tests will be reported as 'failures' on GitHub Checks. Note this does not affect the CI workflow.
+
+## Disabling Skipping For Specific Tests
+
+To ensure that a specific known flakey test is run for your PR, simply include the link to the flakey test's issue anywhere in the body of your PR.
+
+## Still Seeing Flakey Tests?
+
+Sometimes a flakey test will fail all retries. You can try bumping up the number of retries that are attempted by setting the `NUM_RETRIES` env variable.
+
+Some flakey tests are not uncovered by retries. That is, if they fail the first time then every retry will also fail. If you encounter tests like this, please create an issue to track it. If you do so correctly, the flakey test will be disabled until the issue is closed.
+To manually create a flakey test issue, mimic the format of issues created by the FlakeTracker bot. Specifically, make sure to add the `FLAKEY` label as well as labels for the package name and the ci job the test is run in. Also, include the `testID` of the test in the issue title.
+The `testID` can be derived as follows: `jobName -> packageName -> rootDescribeBlockTitle -> ... -> testTitle`. The `testID` is printed each time the test is retried and can be found easily in the logs. Please also include the test error in the issue body.
+
+It is important to note that some flakiness might exist in setup/teardown steps like `before` and `after` hooks. FlakeTracker does not currently address these cases, but you should still create issues to track them!
```

Comment:
Mmmmm, this brings me a question, haha.
If we have 2 flaky test, and it is because of a setup, maybe fixing one of both, will "resolve" the other.
Don't think that is a big problem right now, but maybe we could run all the test batch that lived with a flaky one that was fixed...
For the moment is not that important, but it would be good to add it as a future improvement

Reply:
Yea that's a great point I think that would be a lot of work to implement right now though because the tool doesn't actually have a way of detecting that a test is fixed (the developer the fixes the test is just expected to close the issue so the test is re-enabled). I think a high-impact feature we could add in the future would be checking that a test is fixed. That is, we could have a flag for retrying tests on _success_ (rather than just on failure) and then a test that passes each retry is marked "fixed". This might be non-trivial though, since Mocha and Jest only have native support for retrying _failed_ tests, so we'd have to hack together some other way to retry successful tests. I do think that feature would be really nice to have though when checking whether you've actually fixed a flakey test or not. From there, it would be easy to work it into regular test execution like you describe above.